src/lib.rs

```diff
@@ -240,6 +248,52 @@ impl Opts {
   }
 }
 
+#[derive(Default)]
+pub struct WarningOpts {
+  pub all: bool,
+  pub match_only_vars: bool,
+  pub unused_defs: bool,
+}
+
+impl WarningOpts {
+  pub fn all() -> Self {
+    Self { all: true, match_only_vars: true, unused_defs: true }
+  }
+
+  // TODO(refactor): this kind of code does not look good
+  pub fn from_vec(&mut self, values: Vec<String>) -> Result<(), String> {
+    for value in values {
+      match value.as_ref() {
+        "all" => self.all = true,
```

Comment:
Can we follow the same naming scheme as for selecting compiler passes? `-Wunused-defs` for activating, `-Wno-unused-defs` for disabling the warning.
With default for now being all of them activated.